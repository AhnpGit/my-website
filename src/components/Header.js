import React from "react";

const Header = () => (
    <header>
        <nav>
            <div className="row">
                <img src={require(`../assets/img/logo-white.png`)} alt="Omnifood logo" className="logo" />
                <img src={require(`../assets/img/logo.png`)} alt="Omnifood logo" className="logo-black" />
                <ul className="main-nav">
                    <li><a href="#features">Food delivery</a></li>
                    <li><a href="#works">How it works</a></li>
                    <li><a href="#cities">Our cities</a></li>
                    <li><a href="#plans">Sign up</a></li>
                </ul>
                <a className="mobile-nav-icon"><i className="ion-navicon-round"></i></a>
            </div>
        </nav>
        <div className="hero-text-box">
            <h1>Goodbye junk food.<br />Hello super healthy meals.</h1>
            <a className="btn btn-full" href="">I'm hungry</a>
            <a className="btn btn-ghost" href="">Show me more</a>
        </div>
    </header>
)

export default Header