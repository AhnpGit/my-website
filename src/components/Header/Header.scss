header {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  z-index: map_get($layers, fixed-floating);
  background-color: $trout;
}

.wrapper__header {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;

  height: 60px;
  color: $white;

  @include step-scale('padding-left', $base-spacer, $base-unit, $base-unit);
  @include step-scale('padding-right', $base-spacer, $base-unit, $base-unit);

  @include media-query($medium-up) {
    height: 80px;
  }
}

.header__logo {
  img {
    height: 50px;

    @include media-query($medium-up) {
      height: 70px;
    }
  }
}

.nav__links {
  display: flex;
}

.compactmenu__link {
  @include on-event {
    border: none;
  }
}

.nav__link {
  text-transform: uppercase;
  font-size: 90%;
  text-decoration: none;

  @include on-event {
    border-bottom: none;
  }

  @include media-query($medium-up) {
    color: $mischka;
    border-bottom: 2px solid transparent;
    transition: border-bottom 0.2s;

    @include on-event {
      color: $white;
      border-bottom: 2px solid $tahitiGold;
    }
  }
}

.nav__link:not(:last-child) {
  margin-right: $base-unit;
}

.modal__body {
  ul {
    li {
      line-height: 39px;
      border-bottom: 1px solid $athens-gray;
      padding: 0 $base-unit;
    }
  }
}
