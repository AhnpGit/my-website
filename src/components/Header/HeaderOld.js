// @flow

import { clearFix, rem } from 'polished'
import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'
import CategoryNavigation from '../../../containers/CategoryNavigation'
import CategoryNavigationAndSearch from '../../../containers/CategoryNavigationAndSearch'
import SearchForm from '../../../containers/SearchForm'
import type { DeviceInfo } from '../../modules/deviceInfo'
import type { ShopData } from '../../../modules/shop'
import { base } from '../../../utils/style/branding'
import colors from '../../../utils/style/colors'
import { media, stepScale } from '../../../utils/style/mixins'
import { zIndex } from '../../../utils/style/settings'
import Row from '../../elements/Row/Row'
import Sprite from '../../atoms/Sprite/Sprite'
import Wrapper from '../../atoms/Wrapper/Wrapper'
import MainNavigation from '../../molecules/MainNavigation/MainNavigation'

type Arguments = {
  position?: 'fixed' | 'static',
  navigationMenuIsOpen?: boolean,
  deviceInfo: DeviceInfo,
  shop: ShopData
}

/**
 * Main Header, used on all 'pages' Contains the logo, the main search bar and housed the main navigation
 * The logo is embedded, the search and main navigation are taken in as children
 *
 * The three children are split off to be able to test them separately and in order to be able to make those components reusable
 */

const Header = ({ position = 'fixed', navigationMenuIsOpen = false, deviceInfo, shop }: Arguments) => {

  return (
    <ComponentWrapper position={position}>
      <Component>
        <StyledRow>
          <HeaderTop>
            <TopWrapper>
              <HeaderLogo>
                <LogoSprite name="KrampLogo" />
              </HeaderLogo>
              {!deviceInfo.deviceType.isSmall ? (
                <HeaderSearch>
                  <SearchForm />
                </HeaderSearch>
              ) : ''}
              <HeaderMainNavigation>
                <MainNavigation shop={shop} />
              </HeaderMainNavigation>
            </TopWrapper>
          </HeaderTop>
        </StyledRow>
        <StyledRow>
          <HeaderBottom>
            <BottomWrapper>
              {deviceInfo.deviceType.isSmall ? <CategoryNavigationAndSearch /> : <CategoryNavigation />}
            </BottomWrapper>
          </HeaderBottom>
        </StyledRow>
      </Component>

      {navigationMenuIsOpen && <NavigationMenuBackdrop />}
    </ComponentWrapper>
  )
}

export const ComponentWrapper = styled.div`
  ${ clearFix() };
  position: ${({ position }: { position: 'fixed' | 'static' }) => position};
  top: 0;
  left: 0;
  width: 100%;
  z-index: ${ zIndex.fixedFloating };
`

const Component = styled.div`
  position: relative;
  width: 100%;
  z-index: ${ zIndex.hoverActiveFocus };
`
const StyledRow = styled(Row)`
  margin-bottom: 0;
`

const NavigationMenuBackdrop = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background: rgba(0, 0, 0, 0.3);
  z-index: ${ zIndex.aboveGround };
`
NavigationMenuBackdrop.displayName = 'NavigationMenuBackdrop'

const HeaderTop = styled.div`
  height: ${rem('50px')};
  background-color: ${colors.brightRed};
  color: ${ colors.white };

  ${ media.mediumUp`
    height: ${ rem('70px')};
  ` }
`

const TopWrapper = styled(Wrapper)`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  ${stepScale('padding-left', rem('8px'), base.unit, base.unit)};
  ${stepScale('padding-right', rem('8px'), base.unit, base.unit)}
  height: 100%;
  
  .ie9 & {
    display: block;
  }
`

const LogoSprite = styled(Sprite)`
  width: ${rem('119px')};
  height: ${rem('18px')};

  ${ media.mediumUp `
    width: ${rem('173px')};
    height: ${rem('26px')};
  `}
  
  .ie9 & {
    display: inline-block;
    margin-top: ${ rem('22px') };
  }
`

const HeaderBottom = styled.div`
  background-color: ${colors.redBerry};
  color: ${colors.white};
  height: ${ rem('50px')};

  ${ media.mediumUp`
    height: auto;
  ` }
`

const BottomWrapper = styled(Wrapper)`
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  height: 100%;
  padding: 0 ${rem('5px')};

  ${ media.mediumUp`
      padding: 0;
      justify-content: flex-start;
  ` }
  
  .ie9 & {
    display: block;
  }
`

const HeaderLogo = styled.div`
  display: flex;
  
  .ie9 & {
    width: 33%;
    float: left;
  }
  
  .ie10 & {
    flex: 1 1 auto;
  }
`

const HeaderSearch = styled.div`
  width: 100%;

  ${ media.mediumUp`
    width: ${ rem('250px')};
  ` }
  
  .ie9 & {
    float: left;
    
    ${ media.mediumUp`
      width: 33%;
    ` }
  }
  
  .ie10 & {
    flex: 0 0 250px;
  }
`

const HeaderMainNavigation = styled.div`
  .ie9 & {
    float: right;
  }
  
  .ie10 & {
    flex: 1 1 auto;
    justify-content: flex-end;
  }
`

Header.propTypes = {
  /** dynamic position, used for styleguide only */
  position: PropTypes.string,

  /** Whether or not the navigation menu is currently open. Used to show a backdrop. */
  navigationMenuIsOpen: PropTypes.bool,

  /**
   * Object contains the data of the shop
   */
  shop: PropTypes.object.isRequired
}

export default Header
