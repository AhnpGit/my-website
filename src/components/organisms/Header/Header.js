// @flow

import React from "react"
import PropTypes from 'prop-types'
import Icon from '../../atoms/Icon/Icon'
import type { DeviceInfo } from '../../../utils/types'

type HeaderArguments = {
  deviceInfo: DeviceInfo,
  isOpen: boolean,
  setOpenState: (isOpen: boolean) => mixed
}

/*
 * Header component displays a logo on the left and navigation bar on the right
 */
const Header = ({ deviceInfo, isOpen, setOpenState } : HeaderArguments) => {
  return (
    <header>
      <nav>
        <div className="logo">
          <img src={require('../../../assets/img/logo-white.png')} alt="Omnifood logo" className="logo" />
        </div>
        {deviceInfo.deviceType.isSmall
          ? renderCompactMenu(isOpen, setOpenState)
          : renderNormalNavigation()}
      </nav>
    </header>
  )
}

function renderCompactMenu(isOpen: boolean, setOpenState: (isOpen: boolean) => mixed) {
  return (
    <div className="main__nav__compactmenu">
      <a href="" className="compactmenu__link" onClick={(event) => {
        event.preventDefault()
        setOpenState(!isOpen)
      }}>
        <Icon name="category" />
      </a>
    </div>
  )
}

function renderNormalNavigation() {
  return (
    <ul className="main-nav">
      <li className="nav__item"><a className="main-nav__link" href="#features">Food delivery</a></li>
      <li className="nav__item"><a className="main-nav__link" href="#works">How it works</a></li>
      <li className="nav__item"><a className="main-nav__link" href="#cities">Our cities</a></li>
      <li className="nav__item"><a className="main-nav__link" href="#plans">Sign up</a></li>
    </ul>
  )
}

Header.propTypes = {
  /**
   * Info about the device used here to determine screen size
   */
  deviceInfo: PropTypes.object.isRequired,

  /**
   * Boolean indicates the open state of the navigation menu on small screen
   */
  isOpen: PropTypes.bool.isRequired,

  /**
   * Function which triggers open or close navigation menu on small screen
   */
  setOpenState: PropTypes.func.isRequired
}

export default Header
