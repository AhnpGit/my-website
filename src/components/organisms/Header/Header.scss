.header__wrapper {
  @include clearFix();
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  z-index: map_get($zIndex, fixedFloating);
}

header {
  position: relative;
  width: 100%;
  z-index: map_get($zIndex, hoverActiveFocus);

  .row {
    margin-bottom: 0;
  }
}
