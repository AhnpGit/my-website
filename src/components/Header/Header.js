// @flow

import React from "react"
import PropTypes from 'prop-types'

import Icon from '../../elements/Icon/Icon'
import type { DeviceInfo } from '../../utils/types'

import './Header.scss'
import Modal from '../Modal/Modal'
import ModalHeader from '../Modal/ModalHeader'

import '../Modal/ModalHeader.scss'

type HeaderArguments = {
  deviceInfo: DeviceInfo,
  isOpen: boolean,
  setOpenState: (isOpen: boolean) => mixed
}

/*
 * Header component displays a logo on the left and navigation bar on the right
 */
const Header = ({ deviceInfo, isOpen, setOpenState }: HeaderArguments) => {
  return (
    <div className="header__wrapper">
      <header>
        <div className="header__top">
          <div className="top__wrapper">
            <div className="header__logo">
              <img src={require('../../assets/img/logo-white.png')} alt="Omnifood logo" className="logo" />
            </div>
            {!deviceInfo.deviceType.isSmall
              ? renderNormalNavigation()
              : renderCompactMenu(isOpen, setOpenState)
            }
          </div>
        </div>
      </header>
    </div>
  )
}

function renderCompactMenu(isOpen: boolean, setOpenState: (isOpen: boolean) => mixed) {
  return (
    <nav className="nav__compactmenu">
      <a href="" className="compactmenu__link" onClick={(event) => {
        event.preventDefault()
        setOpenState(!isOpen)
      }}>
        <Icon name="category" />
      </a>
      <Modal
        isOpen={isOpen}
        contentLabel="CategoryNavigationModal Menu"
        onRequestClose={() => setOpenState(false)}
        displayMode="smallScreen"
      >
        <ModalHeader
          onRequestClose={() => setOpenState(false)}
          closeButtonTitle="Close modal"
        >
          Menu
        </ModalHeader>
        <div className="modal__body">
          <ul>
            <li><a className="nav__link" href="#features">Food delivery</a></li>
            <li><a className="nav__link" href="#works">How it works</a></li>
            <li><a className="nav__link" href="#cities">Our cities</a></li>
            <li><a className="nav__link" href="#plans">Sign up</a></li>
          </ul>
        </div>
      </Modal>
    </nav>
  )
}

function renderNormalNavigation() {
  return (
    <nav className="nav__links">
      <a className="nav__link" href="#features">Food delivery</a>
      <a className="nav__link" href="#works">How it works</a>
      <a className="nav__link" href="#cities">Our cities</a>
      <a className="nav__link" href="#plans">Sign up</a>
    </nav>
  )
}

Header.propTypes = {
  /**
   * Info about the device used here to determine screen size
   */
  deviceInfo: PropTypes.object.isRequired,

  /**
   * Boolean indicates the open state of the navigation menu on small screen
   */
  isOpen: PropTypes.bool.isRequired,

  /**
   * Function which triggers open or close navigation menu on small screen
   */
  setOpenState: PropTypes.func.isRequired
}

export default Header
