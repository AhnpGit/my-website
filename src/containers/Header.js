import { connect } from 'react-redux'
import Header from '../components/Header/Header'
import { setOpenState } from '../modules/header'
import type { DeviceInfo, Dispatcher } from '../utils/types'

type HeaderType = {
  deviceInfo: DeviceInfo,
  isOpen: boolean
}

export const mapStateToProps = (state : HeaderType) => ({
  deviceInfo: state.deviceInfo,
  isOpen: state.header.isOpen
})

export const mapDispatchToProps = (dispatch : Dispatcher) => ({
  setOpenState: (isOpen) => {
    dispatch(setOpenState(isOpen))
  }
})

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Header)
