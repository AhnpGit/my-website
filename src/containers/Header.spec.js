import { mapDispatchToProps, mapStateToProps } from './Header'

describe('Header', () => {
  describe('mapStateToProps', () => {
    it('maps the proper fields from the state to the props', () => {
      const state = {
        deviceInfo: {},
        header: {
          isOpen: {}
        }
      }

      const props = mapStateToProps(state)

      expect(props.deviceInfo).toBe(state.deviceInfo)
      expect(props.isOpen).toBe(state.header.isOpen)
    })
  })

  describe('mapDispatchToProps', () => {
    let dispatchWasCalledWith = undefined
    const dispatch = (arg) => dispatchWasCalledWith = arg
    const props = mapDispatchToProps(dispatch)
    it('dispatches an open action', () => {
      props.setOpenState(true)

      expect(dispatchWasCalledWith).toMatchSnapshot()
    })

    it('dispatches a close action', () => {
      props.setOpenState(false)

      expect(dispatchWasCalledWith).toMatchSnapshot()
    })
  })
})
