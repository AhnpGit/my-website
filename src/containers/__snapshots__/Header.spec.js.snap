// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header mapDispatchToProps dispatches a close action 1`] = `
Object {
  "error": undefined,
  "meta": Object {},
  "payload": false,
  "type": "my-website/header/SET_OPEN_STATE",
}
`;

exports[`Header mapDispatchToProps dispatches an open action 1`] = `
Object {
  "error": undefined,
  "meta": Object {},
  "payload": true,
  "type": "my-website/header/SET_OPEN_STATE",
}
`;
